From patchwork Wed Aug 12 02:41:23 2026
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Zhentao Guo via B4 Relay
 <devnull+zhentao.guo.amlogic.com@kernel.org>
X-Patchwork-Id: 155521
Received: from sea.lore.kernel.org ([172.234.253.10])
	by linuxtv.org with esmtps  (TLS1.3) tls
 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
	(Exim 4.96)
	(envelope-from
 <linux-media+bounces-70617-patchwork=linuxtv.org@vger.kernel.org>)
	id 1wtyug-0007MR-35
	for patchwork@linuxtv.org;
	Wed, 12 Aug 2026 02:41:55 +0000
Received: from smtp.subspace.kernel.org (conduit.subspace.kernel.org
 [100.90.174.1])
	by sea.lore.kernel.org (Postfix) with ESMTP id 772633031303
	for <patchwork@linuxtv.org>; Wed, 12 Aug 2026 02:41:39 +0000 (UTC)
Received: from localhost.localdomain (localhost.localdomain [127.0.0.1])
	by smtp.subspace.kernel.org (Postfix) with ESMTP id 2BE03383333;
	Wed, 12 Aug 2026 02:41:36 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=kernel.org header.i=@kernel.org
 header.b="BWbMolET"
X-Original-To: linux-media@vger.kernel.org
Received: from smtp.kernel.org (aws-us-west-2-korg-mail-1.web.codeaurora.org
 [10.30.226.201])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id BFD813806DD;
	Wed, 12 Aug 2026 02:41:35 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org;
 arc=none smtp.client-ip=10.30.226.201
ARC-Seal: i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1786502495; cv=none;
 b=doKYmRnLs7d0g6+G9nS0rvMehuIg9HbyO2a1rBspFFwV8UFa5LGxFsl3BAJ4Vndj6MTj7C+Kbzv4e/P3XhtBEZ4qWqX0o0Odm1KZp+8Tktqg+wZW4ULjhw+JxVp+b/SGue6f9hveFyakOK5/Pn53H9kGmXlYK3rVwSQ5unPmLyY=
ARC-Message-Signature: i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1786502495; c=relaxed/simple;
	bh=PynJ5xo9Sdp/Ik69/y60OZIBerQ/6DJ4h2lEOta72bc=;
	h=From:Date:Subject:MIME-Version:Content-Type:Message-Id:References:
	 In-Reply-To:To:Cc;
 b=kOFEBOKSce6e4FKVu8EVdDRwtaLCsN/XtLaHZBccXQ922sqdzkW5Xmx+o1HK0vfWAsKtq5A/meIA1g5D3VMZWiFdN5HjhDVHVMb3/BwmLZWiRY8rOLmsxa+KvqkCZfRrraEq63nr+XEqMrbxhILm6bqiOcxSxCeJ1K/omK+3/mQ=
ARC-Authentication-Results: i=1; smtp.subspace.kernel.org;
 dkim=pass (2048-bit key) header.d=kernel.org header.i=@kernel.org
 header.b=BWbMolET; arc=none smtp.client-ip=10.30.226.201
Received: by smtp.kernel.org (Postfix) with ESMTPS id 6A64AC2BCC7;
	Wed, 12 Aug 2026 02:41:35 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=kernel.org;
	s=k20201202; t=1786502495;
	bh=PynJ5xo9Sdp/Ik69/y60OZIBerQ/6DJ4h2lEOta72bc=;
	h=From:Date:Subject:References:In-Reply-To:To:Cc:Reply-To:From;
	b=BWbMolET+BUv+HCNiYSywfCMTafMv8EbTu0ttnqa/PNdzATzYBBf4YeM4WKsEDDa0
	 HrAmpp3S79mhFqgPiZgBmjS2ztfwAIYgV4VpC0sgtoWCN1DF7ngjKh/vCPCgrju3Xz
	 Hkyp0Mi/02tM34/G2qs+T92MqTZ89rh3WJJfKIg5Fws1TqeLLTcn42LrCVpeeEjVeL
	 DZ8ioJpfLv2n+pavUKjrihHzeXLRLt9ODk+4CYz9j/LAIMpUBSF0I0QkEqgo7f1faR
	 DEj2EICD53DFsFcdjXuYf9kK+y3KpgNZI9qBK8Zg8ssKpsUTl2wHjC7sp43Ud/D6Vw
	 C5bLQd9OPIsgA==
Received: from aws-us-west-2-korg-lkml-1.web.codeaurora.org
 (localhost.localdomain [127.0.0.1])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 48316C5AC67;
	Wed, 12 Aug 2026 02:41:35 +0000 (UTC)
From: Zhentao Guo via B4 Relay <devnull+zhentao.guo.amlogic.com@kernel.org>
Date: Wed, 12 Aug 2026 10:41:23 +0800
Subject: [PATCH v7 1/6] firmware: meson: sm: video firmware loading via
 secure monitor
Precedence: bulk
X-Mailing-List: linux-media@vger.kernel.org
List-Id: <linux-media.vger.kernel.org>
List-Subscribe: <mailto:linux-media+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-media+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Message-Id: <20260812-b4-s4-vdec-upstream-v7-1-3fae085861bd@amlogic.com>
References: <20260812-b4-s4-vdec-upstream-v7-0-3fae085861bd@amlogic.com>
In-Reply-To: <20260812-b4-s4-vdec-upstream-v7-0-3fae085861bd@amlogic.com>
To: Neil Armstrong <neil.armstrong@linaro.org>,
 Kevin Hilman <khilman@baylibre.com>, Jerome Brunet <jbrunet@baylibre.com>,
 Martin Blumenstingl <martin.blumenstingl@googlemail.com>,
 Mauro Carvalho Chehab <mchehab@kernel.org>, Rob Herring <robh@kernel.org>,
 Krzysztof Kozlowski <krzk+dt@kernel.org>,
 Conor Dooley <conor+dt@kernel.org>
Cc: linux-arm-kernel@lists.infradead.org, linux-amlogic@lists.infradead.org,
 linux-kernel@vger.kernel.org, linux-media@vger.kernel.org,
 devicetree@vger.kernel.org, Zhentao Guo <zhentao.guo@amlogic.com>
X-Mailer: b4 0.14.3
X-Developer-Signature: v=1; a=ed25519-sha256; t=1786502492; l=670;
 i=zhentao.guo@amlogic.com; s=20251024; h=from:subject:message-id;
 bh=ioNPQiZsvUfM7WHdEfF4xb07l7H3o3hVIdC2XZUtO1g=;
 b=eK1kDLPi+TZ0ozmq9apoa6V7z/Tacdapq4Ey5p5r/Sn47u2Ng6Q6TZMe7fL2KhAE4D5HqpoDk
 Or7Tr9aXEvzDGucXn6s0rGvzAFbylCH9EbAjRSx6HrMqVTJnk2bFngM
X-Developer-Key: i=zhentao.guo@amlogic.com; a=ed25519;
 pk=5yfDKrjreXwcAoEUsdtWafy6YN500upXp/CgtnXjLVU=
X-Endpoint-Received: by B4 Relay for zhentao.guo@amlogic.com/20251024 with
 auth_id=555
X-Original-From: Zhentao Guo <zhentao.guo@amlogic.com>
Reply-To: zhentao.guo@amlogic.com
X-LSpam-Score: -7.8 (-------)
X-LSpam-Report: No,
 score=-7.8 required=5.0
 tests=ARC_SIGNED=0.001,ARC_VALID=-0.1,BAYES_00=-1.9,DKIMWL_WL_HIGH=-1,DKIM_SIGNED=0.1,DKIM_VALID=-0.1,DKIM_VALID_AU=-0.1,DMARC_PASS=-0.001,MAILING_LIST_MULTI=-1,RCVD_IN_VALIDITY_CERTIFIED=-3,RCVD_IN_VALIDITY_RPBL=1.31,RCVD_IN_VALIDITY_SAFE=-2,SPF_HELO_NONE=0.001,SPF_PASS=-0.001
 autolearn=ham autolearn_force=no

From: Zhentao Guo <zhentao.guo@amlogic.com>

Add SM_LOAD_VIDEO_FW to the secure monitor command enum
to allow decoder drivers to load firmware through the meson_sm
interface.

Signed-off-by: Zhentao Guo <zhentao.guo@amlogic.com>
---
 include/linux/firmware/meson/meson_sm.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/firmware/meson/meson_sm.h b/include/linux/firmware/meson/meson_sm.h
index 3ebc2bd9a976..3b90b49d0552 100644
--- a/include/linux/firmware/meson/meson_sm.h
+++ b/include/linux/firmware/meson/meson_sm.h
@@ -15,6 +15,7 @@ enum {
 	SM_THERMAL_CALIB_READ,
 	SM_A1_PWRC_SET,
 	SM_A1_PWRC_GET,
+	SM_LOAD_VIDEO_FW,
 };
 
 struct meson_sm_firmware;
